src/main/resources/test_files/runtime-spark/LocalClauses/WhereClauseLocal6

```diff
@@ -0,0 +1,5 @@
+(:JIQS: ShouldRun; Output="" :)
+for $i in ()
+where $j := $i + 1
```

Comment:
The test actually didn't succeed, as a little trick to deactivate the (buggy) tests temporarily, I just remove the ".iq" postfix.

>for $i in ()
>where $j eq $i + 1
>return $i

In the current state the test case given above causes "(no URI):2,7: static error [err:XPST0008]: "j": undeclared variable" both in Zorba and sparksoniq. Shall we add this as a "ShouldCrash" test case?

Reply:
I missed that too! Let's replace $j with an integer. I forgot about this. Of course, you are welcome to add one more test case with a ShouldCrash setting. Thanks!

Reply:
I will add the mentioned(integer) test case and then merge the PR. Before that, I have one last question given below:

The following test annotation fails with the assertion given below:
(:JIQS: ShouldCrash; ErrorCode="XPST0008"; ErrorMetadata="LINE:3:COLUMN:6:" :)
for $i in ()
where $j eq $i + 1
return $i

java.lang.AssertionError: Program did not compile when expected to.
Error output: Error [err: XPST0008]LINE:3:COLUMN:6:Uninitialized variable reference: j

It appears that this test case causes a semantic exception which later hits a shouldCompile() check that always returns true, effectively causing the given exception. If semantic exceptions are a part of test cases handled at runtime, a fix may be needed here. If they are related to an earlier stage, this can be ignored. Should I create an issue on this?

Reply:
Indeed, this program should parse, but static analysis should then spot the unbound variable. Can you open an issue?